### PR TITLE
Allow only admins to add and delete users

### PR DESCRIPTION
**Pull request by limxinyuuu**
This change creates a users/index page that only admin users are able to access. On the users/index page, admin users are able to add a new user or delete existing users.

Note: An admin column has been added to the users table. Run `rake db:migrate` to migrate. The seed file is also updated so that `dawn@highpoint.com` is an admin.

Screenshot:
image

---

**Before submitting, check that:**

 - [ ] You have added (passing) tests for your code.
 - [x] You have written good* commit messages.
 - [ ] You have squashed relevant commits together.
 - [x] You have ensured that RuboCop is passing.
 - [x] Your PR relates to one subject, with a clear title and
       description using complete, grammatically correct sentences.

---

If your change contains models, ensure that:

 - [ ] You have included an updated screen shot of the ERD.
 - [ ] You have added database seeds for the model.
 - [ ] You have added working factories for the model.
 - [ ] Your validations have corresponding database constraints.

---

If your change contains views, ensure that:

 - [x] You have included a screenshot of the new view.

---

*If you do not know what makes a good commit message, or why it
is important, please refer to these resources: [1][1], [2][2], [3][3].

[1]: https://robots.thoughtbot.com/5-useful-tips-for-a-better-commit-message
[2]: http://chris.beams.io/posts/git-commit/
[3]: https://github.com/blog/1943-how-to-write-the-perfect-pull-request

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
+    else
+      user.destroy
+      redirect_to users_path
+      flash[:success] = "User successfully deleted."
+    end
+  end
+
+  private
+
+  def admin_user
```
Comment (Drenmi) [inline app/controllers/users_controller.rb]: This method name indicates that the method returns an admin user, which doesn't seem to be the case. 😅 How about `#require_admin_privileges!`

File: app/views/users/index.html.slim [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          tr
+            td = user.email
+            td = l user.created_at.to_date, format: user.created_at.year == Date.today.year ? :short : :default
+            td
+              = link_to user, method: :delete, data:{ confirm: "Delete this user?"} do
+                i.fa.fa-trash.fa-lg
+ 
+.text-xs-center
+  .row
+    = will_paginate @users
```
Comment (Drenmi) [inline app/views/users/index.html.slim]: Missing a column here? 😀

File: config/routes.rb [hunk 3]
```diff
@@ -7,7 +7,7 @@
   end
   resources :donors, only: [:index, :show, :update, :edit]
   resources :events, only: [:create, :index, :show, :update, :edit]
-  resources :users, only: [:create, :show]
+  resources :users, only: [:create, :show, :index, :destroy]
```
Comment (Drenmi) [inline config/routes.rb]: If possible, let's try to keep the actions in the same order for the different actions. 😀

**Comment by limxinyuuu** [comment]
@Drenmi Realised that the schema changes in the pull request accidentally overrode those in the master :'( How do I fix the schema?

**Pull request merged**